apps/dashboard/app/models/batch_connect/session.rb

```diff
@@ -226,6 +225,8 @@ def save(app:, context:, format: nil)
     # @param context [Object] context available when rendering staged files
     # @return [Boolean] whether staged successfully
     def stage(root, context: nil)
+      staged_root.tap { |p| p.mkpath unless p.exist? }
```

Comment:
Now we'll only create the `staged_root` when `stage` is called. This is so that errors in `submit.yml.erb` (that now has access to staged_root) won't create empty directories as folks triage what's wrong in their erb files.